### ISSUE TITLE
Vault Agent: Cannot use auto_auth with template without specifying a sink

### ISSUE THREAD

**Issue by adongy** [hashicorp/vault#7988]
**Is your feature request related to a problem? Please describe.**
I'm trying to use the Vault agent to solve authentication and templating of secrets at the same time.

My current solution is to use Vault agent auto auth and consul-template to do it:

- Vault auth_auth writes to a sink
- consul-template consumes the secret and templates the config file

Vault agent recently provided the `template` block that mirrors consul-template, and I'd like to use it to avoid loading two binaries for this.

However, with this sample config, I get the following error:
`Error loading configuration from vault-agent.hcl: auto_auth requires at least one sink or cache.use_auto_auth_token=true`

```hcl
exit_after_auth = true
pid_file = "/run/vault-agent.pid"

auto_auth {
  method "aws" {
    mount_path = "auth/aws"
    config = {
      type = "iam"
      role = "role"
      header_value = "vault.example.com"
    }
  }
}

template {
  contents = <<EOF
{{with secret "secret/path/to/secret"}}
{{range $k, $v := .Data.data}}
{{ $k }}={{ $v }}
{{end}}
{{end}}
EOF
  destination = "/run/app-env"
  error_on_missing_key = true
}
```

Adding a sink does solve the error, but I would rather not write a token that I don't use.

**Describe the solution you'd like**
If `template` is declared, disable the above error

**Describe alternatives you've considered**
Either adding a sink or using consul-template solves this.

**Explain any additional use-cases**

**Additional context**

Thanks!

**Comment by cidrick** [comment]
A separate, but similar, use case for me is to leverage the vault agent for its auto-auth and caching capabilities only, with no need to share the vault token via a shared volume. Writing the token to the container's filesystem exposes the token to anyone who can obtain a shell in the container. 

Since we have no need for a file-based sink at all, it'd be more secure to just keep the token in-memory.

**Comment by mikegreen** [comment]
@cidrick a workaround to not write the sink requires a dummy listener, but here is that syntax for the config in case someone else needs this functionality prior to it being released:
```# this is a workaround to not write a sink file containing 
# a token file that could be a security risk
#
listener "unix" {
         address = "foo.txt"
         tls_disable = true
}

cache {
  use_auto_auth_token = true
}
```

**Comment by ncabatoff** [comment]
Fixed in #8812.
